invenio/modules/records/models.py

```diff
@@ -159,6 +159,30 @@ class RecordMetadata(db.Model):
     record = db.relationship(Record, backref='record_json')
 
 
+class RecordMetadataHistory(db.Model):
+
+    """Represents the revisions of a record if any."""
+
+    __tablename__ = 'record_json_history'
+
+    id = db.Column(
+        db.MediumInteger(8, unsigned=True),
+        db.ForeignKey(Record.id),
+        primary_key=True,
+        nullable=False
+    )
+
+    revision = db.Column(
+        db.DateTime,
+        primary_key=True,
+        nullable=False
+    )
+
+    json = db.Column(db.JSON, nullable=False)
+
+    record = db.relationship(Record, backref='record_json_history')
```

Comment:
If we go with ``RecordHistory`` then perhaps we can call the back reference just ``history``?